pkg/agent/proxy/proxier.go

```diff
@@ -1141,10 +1148,20 @@ func NewProxier(
 		serviceHealthServer = healthcheck.NewServiceHealthServer(hostname, nil, nodePortAddressesString)
 	}
 
+	nonHeadlessServiceSelector, _ := labels.NewRequirement(corev1.IsHeadlessService, selection.DoesNotExist, nil)
```

Comment:
Please check my comment in https://github.com/antrea-io/antrea/pull/4973#discussion_r1235238563